client/my-sites/domains/domain-management/list/domain-only.jsx

```diff
@@ -6,27 +6,52 @@ import Illustration from 'calypso/assets/images/domains/domain.svg';
 import QuerySiteDomains from 'calypso/components/data/query-site-domains';
 import EmptyContent from 'calypso/components/empty-content';
 import { canCurrentUserCreateSiteFromDomainOnly } from 'calypso/lib/domains';
+import { transferStatus, type as domainTypes } from 'calypso/lib/domains/constants';
 import { hasGSuiteWithUs } from 'calypso/lib/gsuite';
 import { hasTitanMailWithUs } from 'calypso/lib/titan';
 import { domainManagementEdit, createSiteFromDomainOnly } from 'calypso/my-sites/domains/paths';
 import { emailManagement } from 'calypso/my-sites/email/paths';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import getCurrentRoute from 'calypso/state/selectors/get-current-route';
 import getPrimaryDomainBySiteId from 'calypso/state/selectors/get-primary-domain-by-site-id';
+import { getDomainsBySiteId } from 'calypso/state/sites/domains/selectors';
 import { getSiteSlug } from 'calypso/state/sites/selectors';
 
 import './domain-only.scss';
 
 const DomainOnly = ( {
 	currentRoute,
 	primaryDomain,
+	domains,
 	hasNotice,
 	recordTracks,
 	siteId,
 	slug,
 	translate,
 } ) => {
 	/* eslint-disable wpcalypso/jsx-classname-namespace */
+	const domain = domains.find( ( item ) => item.name === slug );
```

Comment:
can you please elaborate a bit about what we are doing here?

Reply:
The code looks for the domain object associated with the slug (which is identical to the domain they are transferring).

Using the domain object, there are properties on the domain type and transfer status, which is used to determine whether to show that "egao.dev is not ready yet" placeholder, instead of the infinitely-spinning loading placeholder.